guides/fcm-flutter-novu/how-to-send-push-notifications-to-flutter-apps-with-fcm-using-Novu.mdx

```diff
@@ -0,0 +1,735 @@
+          _lastMessage = 'Received a notification message:'
+              '\nTitle=${message.notification?.title},'
+              '\nBody=${message.notification?.body},'
+              '\nData=${message.data}';
+        } else {
+          _lastMessage = 'Received a data message: ${message.data}';
+        }
+      });
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: Text(widget.title),
+      ),
+      body: Center(
+        child: Column(
+          mainAxisAlignment: MainAxisAlignment.center,
+          children: <Widget>[
+            Text('Last message from Firebase Messaging:',
+                style: Theme.of(context).textTheme.titleLarge),
+            Text(_lastMessage, style: Theme.of(context).textTheme.bodyLarge),
+          ],
+        ),
+      ),
+    );
+  }
+}
+```
+_lib/main.dart_
+
+There are 4 sections to pay attention to:
+
+<Steps>
+  <Step title="The Request permission code block">
+    This shows the notification permission prompt to the user the first time the user interacts with the app. If the user allows it, then we can get a token.
+
+    ```dart
+    // Request permission
+    final messaging = FirebaseMessaging.instance;
+
+    // Web/iOS app users need to grant permission to receive messages
+    final settings = await messaging.requestPermission(
+      alert: true,
+      announcement: false,
+      badge: true,
+      carPlay: false,
+      criticalAlert: false,
+      provisional: false,
+      sound: true,
+    );
+    ```
+
+  </Step>
+  <Step title="Registering and obtaining the FCM token for the device">
+    A device's token is needed for the specific device to receive messages. The `getToken()` returns the registration token for the app instance on the device.
+
+    ```dart
+      String? token = await messaging.getToken();
+
+      if (kDebugMode) {
+        print('Registration Token=$token');
+      }
+    ```
+  </Step>
+  <Step title="The foreground message handler">
+    This handler listens to when a push notification is fired from FCM to the device. If the app is in foreground, then it prints out the notification title, body, messageId and data properties to the console.
+
+    ```dart
+      FirebaseMessaging.onMessage.listen((RemoteMessage message) {
+        if (kDebugMode) {
+          print('Handling a foreground message: ${message.messageId}');
+          print('Message data: ${message.data}');
+          print('Message notification: ${message.notification?.title}');
+          print('Message notification: ${message.notification?.body}');
+        }
+
+        _messageStreamController.sink.add(message);
+      });
+    ```
+  </Step>
+  <Step title="The background message handler">
+    This handler listens to when a push notification is fired from FCM to the device. If the app is in background, then it prints out the notification title, body, messageId and data properties to the console.
```

Comment:
1. This handler listens to when a push notification is fired from FCM to the device => This handler is triggered when a push notification is fired from FCM to the device.
2. If app is in background => if app is in the background
3. messageId and data => messageId, and data